src/FormBuilder/BlockModels/DonationAmountBlockModel.php

```diff
@@ -60,23 +60,28 @@ public function getLabel(): string
     }
 
     /**
+     * @unreleased Changed the return type to an array of OptionsProps
      * @since 3.0.0
      *
-     * @return float[]
+     * @return array ['label' => string, 'value' => string, 'checked' => bool][]
      */
     public function getLevels(): array
     {
         return array_map(static function($level) {
-            return (float)filter_var($level, FILTER_SANITIZE_NUMBER_FLOAT,FILTER_FLAG_ALLOW_FRACTION);
+            return [
+                'label' => (string)filter_var($level['label'] ?? '', FILTER_SANITIZE_STRING),
```

Comment:
@pauloiankoski heads up `FILTER_SANITIZE_STRING` is [deprecated](https://www.php.net/manual/en/filter.constants.php#constant.filter-sanitize-string) in php 8.1 - they are recommending to use [htmlspecialchars()](https://www.php.net/manual/en/function.htmlspecialchars.php) instead

Reply:
Resolved: https://github.com/impress-org/givewp/commit/2f368c85e5774c4b28e0b9346d5b4a2981d17107